### ISSUE TITLE
VSCode support for bad chars

### ISSUE THREAD

**Issue by lonix1** [nhoizey/vscode-gremlins#225]
VSCode now has [built-in support](https://code.visualstudio.com/updates/v1_63#_unicode-highlighting) for detecting "bad" characters.

Is this extension obsolete, or does it offer more features? I've been using it for a while and unsure whether to keep it or not?

**Comment by nhoizey** [comment]
@lonix1 good catch,thanks! 👍

I guess this extension can still be useful, at least with some characters VSCode doesn't plan to highlight.

We have to define which characters should be removed from the extension.

**Comment by lonix1** [comment]
You're right - it would be good not to redo the work vscode already does, and handle bad chars that vscode doesn't catch. But how would we even know how to detect that?

**Comment by kaushalyap** [comment]
I am also interested in this

**Comment by TheSench** [comment]
There are some characters that this extension highlights by default that VSCode does not (`–` en dash, for instance).  You can get VSCode to highlight those if you turn on `editor.unicodeHighlight.nonBasicASCII` though.

As far as features that this extension provides that VSCode does not, I can think of the following:
1. Ability to display these characters in the **Problems** pane - currently, VSCode's built-in functionality will highlight them and display them in the gutter
2. Ability to customize the list of highlighted characters (double-edged sword, as creating a custom list means you lose out on updates to the default list)
3. Ability to adjust the severity of characters
4. Language-specific settings (e.g. allow special characters in `.txt` files or a different list in `.md` files)
5. "Zap Gremlins" was on the roadmap, allowing for quick removal of special characters

**Comment by lonix1** [comment]
@TheSench Thank you for the detailed overview. In fact in might be a good idea to add that list to the docs, because I'm sure many users will wonder about this now.

**Comment by AffluentOwl** [comment]
I want to use this extension to highlight certain characters I care about. Such as the letter "A", in certain situations where it isn't commonly invalid, just in the files / file formats I care about.

**Comment by villu164** [comment]
I just spent a lot of time trying to figure out, why one worked and the other didn't, just to find the old "en-dash vs hyphen" to be the gremlin. 
```sh
nmap -p 139,445 -Pn -script smb-enum* 127.0.0.1
```

```sh
nmap -p 139,445 -Pn –script smb-enum* 127.0.0.1
```

It didn't blow up, it just didn't work 😫 

True, the `editor.unicodeHighlight.nonBasicASCII = true` does highlight the en-dash, but then again, a lot of other things aswell, like

gremlin disabled and setting enabled
<img width="233" alt="gremlin disabled and setting enabled" src="https://user-images.githubusercontent.com/998682/185739999-8d24c882-d8e9-4118-abd3-7e1ee26433ef.png">

nothing enabled
<img width="234" alt="nothing enabled" src="https://user-images.githubusercontent.com/998682/185740012-9fbcc964-39d3-46c2-a872-d2196c59f00a.png">

gremlin enabled and setting disabled
<img width="237" alt="gremlin enabled and setting disabled" src="https://user-images.githubusercontent.com/998682/185740025-6797ef67-f81f-44b5-b843-436370f3bfc8.png">
